auto resend tick votes if tick gets stuck too long

krypdkat opened the pull request.

J0ET0M commented:
logic looks good to me.

maybe we should restrict the check to be done in MAIN state only

krypdkat commented:
@J0ET0M addressed all concerns. New logic: 
```
// Resend vote from this node when:
// - timeout
// - on the last tick, this node was in MAIN mode
```

J0ET0M commented:
looks good to me. thanks!
